sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/ViewDescriptorFactory.scala

```diff
@@ -176,51 +179,76 @@ private[impl] object ViewDescriptorFactory extends ComponentDescriptorFactory {
-          case None => previousEntityClass = Some(entityClass)
-        }
+      def invalidComponentException(method: Method, extraMessage: String) = {
+        throw InvalidComponentException(
+          s"Method [${method.getName}] annotated with '@Subscribe.EventSourcedEntity' should either receive " +
+          "a single parameter of one of the event types or " +
+          s"two ordered parameters  of type [${tableType.getName}] and an event type. $extraMessage")
+      }
 
-        method.getParameterTypes.toList match {
-          case params if params.size > 2 =>
-            invalidComponentException(
-              method,
-              s"Subscription method should have only one parameter, found ${params.size}")
-          case _ => // happy days, dev did good with the signature
+      import ReflectionUtils.methodOrdering
+      var previousEntityClass: Option[Class[_]] = None
+
+      methods.sorted // make sure we get the methods in deterministic order
+        .map { method =>
+          // validate that all updates return the same type
+          val entityClass = method.getAnnotation(classOf[Subscribe.EventSourcedEntity]).value().asInstanceOf[Class[_]]
+
+          previousEntityClass match {
+            case Some(`entityClass`) => // ok
+            case Some(other) =>
+              throw InvalidComponentException(
+                s"All update methods must return the same type, but [${method.getName}] returns [${entityClass.getName}] while a previous update method returns [${other.getName}]")
+            case None => previousEntityClass = Some(entityClass)
+          }
+
+          method.getParameterTypes.toList match {
+            case params if params.size > 2 =>
+              invalidComponentException(
+                method,
+                s"Subscription method should not have more than 2 parameters, found ${params.size}")
+            case _ => // happy days, dev did good with the signature
+          }
         }
+      methods
+    }
 
-        // event sourced or topic subscription updates
-        val methodOptionsBuilder = kalix.MethodOptions.newBuilder()
-        methodOptionsBuilder.setEventing(eventingInForEventSourcedEntity(method))
-        addTableOptionsToUpdateMethod(tableName, tableProtoMessageName, methodOptionsBuilder, true)
+    def getEventing(method: Method, component: Class[_]): Eventing =
+      if (hasEventSourcedEntitySubscription(component)) eventingInForEventSourcedEntity(component)
+      else eventingInForEventSourcedEntity(method)
 
-        KalixMethod(SubscriptionServiceMethod(method))
-          .withKalixOptions(methodOptionsBuilder.build())
+    def findIgnores(method: Method, component: Class[_]): Boolean =
+      if (hasEventSourcedEntitySubscription(component)) findIgnore(component)
+      else findIgnore(method)
```

Comment:
But we don't support it as method annotation, so why we need it?

I see that we are using it to construct `SubscriptionServiceMethod`, but that will probably generate gRPC method with a invalid annotation.

Reply:
now you know we don't pass it to the proxy